pca_test.go

```diff
@@ -0,0 +1,282 @@
+				5.5, 4.2, 1.4, 0.2,
+				4.9, 3.1, 1.5, 0.1,
+				5.0, 3.2, 1.2, 0.2,
+				5.5, 3.5, 1.3, 0.2,
+				4.9, 3.1, 1.5, 0.1,
+				4.4, 3.0, 1.3, 0.2,
+				5.1, 3.4, 1.5, 0.2,
+				5.0, 3.5, 1.3, 0.3,
+				4.5, 2.3, 1.3, 0.3,
+				4.4, 3.2, 1.3, 0.2,
+				5.0, 3.5, 1.6, 0.6,
+				5.1, 3.8, 1.9, 0.4,
+				4.8, 3.0, 1.4, 0.3,
+				5.1, 3.8, 1.6, 0.2,
+				4.6, 3.2, 1.4, 0.2,
+				5.3, 3.7, 1.5, 0.2,
+				5.0, 3.3, 1.4, 0.2,
+			}).T(),
+			k: 3,
+			wantPC: mat64.NewDense(50, 3, []float64{
+				-0.1460045, -0.026345338, 0.030540457,
+				-0.1360308, 0.173498400, 0.080837321,
+				-0.1338195, -0.009169147, 0.037035810,
+				-0.1279703, 0.067555699, -0.126221445,
+				-0.1443286, -0.104408539, -0.029350370,
+				-0.1493778, -0.074299623, -0.058439654,
+				-0.1295216, -0.114442671, -0.058448914,
+				-0.1412960, 0.023838362, -0.056736653,
+				-0.1219751, 0.088299660, -0.072835231,
+				-0.1382634, 0.152085262, -0.084122935,
+				-0.1546217, -0.019878976, 0.012748139,
+				-0.1349115, -0.004041140, -0.203904591,
+				-0.1358361, 0.149186669, -0.019439756,
+				-0.1261483, -0.066488387, 0.030531197,
+				-0.1713692, -0.143219751, 0.310173366,
+				-0.1642547, -0.300015329, 0.078817669,
+				-0.1539403, -0.180464207, 0.245480926,
+				-0.1435772, -0.029243931, 0.095223637,
+				-0.1587249, 0.061082818, 0.007955787,
+				-0.1452746, -0.155261416, -0.053647302,
+				-0.1495025, 0.185761948, -0.066321357,
+				-0.1419013, -0.107307131, 0.035332809,
+				-0.1384033, -0.319414417, 0.132194623,
+				-0.1334088, 0.146288077, 0.045243424,
+				-0.1314896, 0.075582298, -0.431845026,
+				-0.1363714, 0.253791016, -0.035529073,
+				-0.1353008, 0.044582323, -0.003350438,
+				-0.1474858, 0.027406132, -0.009845791,
+				-0.1476805, 0.051717863, 0.090431285,
+				-0.1303976, 0.070454291, -0.190904625,
+				-0.1320736, 0.148517492, -0.131013797,
+				-0.1469291, 0.126882471, 0.215005292,
+				-0.1555890, -0.274812538, -0.220310558,
+				-0.1631141, -0.273474183, 0.002837524,
+				-0.1382634, 0.152085262, -0.084122935,
+				-0.1428259, 0.045920678, 0.219797644,
+				-0.1576329, 0.055954810, 0.248896188,
+				-0.1382634, 0.152085262, -0.084122935,
+				-0.1240617, 0.010905637, -0.021152017,
+				-0.1439179, 0.051048685, -0.021142756,
+				-0.1420959, -0.082995400, 0.135609885,
+				-0.1176345, 0.391187507, 0.249203578,
+				-0.1259537, -0.090800118, -0.069745879,
+				-0.1313922, -0.012067739, 0.101718990,
+				-0.1382848, -0.051995424, -0.292884702,
+				-0.1309815, 0.143389484, 0.109926604,
+				-0.1465612, -0.125821677, -0.194310626,
+				-0.1300569, -0.009838325, -0.074538231,
+				-0.1519997, -0.047089299, -0.022845757,
+				-0.1414906, 0.048150093, 0.043540423,
+			}),
+			wantProj: nil,
+			wantVars: []float64{222.78428394075226, 0.8252354865184671, 0.2821472393958653},
+			epsilon:  1e-6, // Reduced precision because PC vectors are obtained from R by copy/paste.
+		},
+	} {
+		pc, proj, vars, ok := PrincipalComponents(test.data, nil, test.k)
```

Comment:
We should have a test with weighted data. I can add that later if you'd prefer given that the addition is my fault :-).

In either event, it's better for weights to be an element in test rather than a nil literal.